File: flutter_frontend/lib/models/reaction.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:universe/models/user.dart';

class Reaction {
  User? user;
  final String userReference;
  final String reactionType;
  final Timestamp reactionDate;

  Reaction({
    required this.userReference,
    required this.reactionType,
    required this.reactionDate,
  });

  factory Reaction.fromFirestore(
    DocumentSnapshot<Map<String, dynamic>> snapshot,
    SnapshotOptions? options,
  ) {
    final data = snapshot.data();
    return Reaction(
      userReference: data?['userReference'],
      reactionType: data?['type'],
      reactionDate: data?['reactionDate'],
    );
  }

  Map<String, dynamic> toFirestore() {
    return {
      'userReference': userReference,
      'type': reactionType,
      'reactionDate': reactionDate,
    };
  }
}

File: Universe_Backend/Data/Models/Reaction.cs
namespace Universe_Backend.Data.Models;

public class Reaction
{
    public required int Id { get; set; }
    public required string UserId { get; set; }
    public required string ReactionType { get; set; }
    public DateTime ReactionDate { get; set; } = DateTime.Now;

    public virtual User? User { get; set; }
}